Make adapter info into `frontMatter`-powered snippet table (#2040)
fishtown-analytics/docs.getdbt.com: website/docs/reference/warehouse-setups/azuresynapse-setup.md

<ul>
    <li><strong>Maintained by</strong>: {frontMatter.meta.maintained_by}</li>
    <li><strong>Authors</strong>: {frontMatter.meta.authors}</li>
    <li><strong>GitHub repo</strong>: <a href={`https://github.com/${frontMatter.meta.github_repo}`}>{frontMatter.meta.github_repo}</a><a href={`https://github.com/${frontMatter.meta.github_repo}`}><img src={`https://img.shields.io/github/stars/${frontMatter.meta.github_repo}?style=for-the-badge`}/></a></li>
    <li><strong>PyPI package</strong>: <code>{frontMatter.meta.pypi_package}</code> <a href={`https://badge.fury.io/py/${frontMatter.meta.pypi_package}`}><img src={`https://badge.fury.io/py/${frontMatter.meta.pypi_package}.svg`}/></a></li>
    <li><strong>Slack channel</strong>: <a href={frontMatter.meta.slack_channel_link}>{frontMatter.meta.slack_channel_name}</a></li>
    <li><strong>Supported dbt Core version</strong>: {frontMatter.meta.min_core_version} and newer</li>
    <li><strong>dbt Cloud support</strong>: {frontMatter.meta.cloud_support}</li>
    <li><strong>Minimum data platform version</strong>: {frontMatter.meta.min_supported_version}</li>
    </ul>

<h2> Installing {frontMatter.meta.pypi_package} </h2>

> The easiest way to install the adapter is to use pip:

Reviewer: ```suggestion
pip is the easiest way to install the adapter:
```

<code>pip install {frontMatter.meta.pypi_package}</code>

> <p>You don't need to install dbt separately. Installing <code>{frontMatter.meta.pypi_package}</code> will also install <code>dbt-core</code> and any other dependencies.</p>

Reviewer: ```suggestion
<p>Installing <code>{frontMatter.meta.pypi_package}</code> will also install <code>dbt-core</code> and any other dependencies.</p>
```

<h2> Configuring {frontMatter.meta.pypi_package} </h2>

<p>For {frontMatter.meta.platform_name}-specifc configuration please refer to <a href={frontMatter.meta.config_page}>{frontMatter.meta.platform_name} Configuration</a> </p>

<p>For further info, refer to the GitHub repository: <a href={`https://github.com/${frontMatter.meta.github_repo}`}>{frontMatter.meta.github_repo}</a></p>

:::info Dedicated SQL only

> Azure Synapse offers both Dedicated SQL Pools and Serverless SQL Pools.

Reviewer: ```suggestion
Azure Synapse offers both Dedicated and Serverless SQL Pools.
```

> **Only Dedicated SQL Pools are supported by this adapter. If you really insist on using serverless pools, check out the neglected, experimental project: [dbt-synapse-serverless](https://github.com/dbt-msft/dbt-synapse-serverless)** 

Reviewer: ```suggestion
**This adapter supports only Dedicated SQL Pools. If you must use Serverless pools, there is an experimental project that may help, but it has not been updated recently: [dbt-synapse-serverless](https://github.com/dbt-msft/dbt-synapse-serverless)** 
```

:::

### Prerequisites

> On Ubuntu make sure you have the ODBC header files before installing

Reviewer: ```suggestion
On Ubuntu, make sure you have the ODBC header files before installing:
```

    sudo apt install unixodbc-dev

Download and install the [Microsoft ODBC Driver 17 for SQL Server](https://docs.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver15)

### Authentication methods

> This adapter is based on the adapter for Microsoft SQL Server.
> Therefor, the same authentication methods are supported.

Reviewer: ```suggestion
This adapter is based on the adapter for Microsoft SQL Server, and the same authentication methods are supported.
```

> The configuration is the same except for 1 major difference:

Reviewer: ```suggestion
The configuration is the same except for one significant difference:
```

> instead of specifying `type: sqlserver`, you specify `type: synapse`.

Reviewer: ```suggestion
Specify `type: synapse` rather than `type: sqlserver`. 
```

Example:

<File name='profiles.yml'>

```yaml
your_profile_name:
  target: dev
  outputs:
    dev:
      type: synapse
      driver: 'ODBC Driver 17 for SQL Server' # (The ODBC Driver installed on your system)
      server: workspacename.sql.azuresynapse.net # (Dedicated SQL endpoint of your workspace here)
      port: 1433
      schema: schema_name
      user: username
      password: password
```

</File>

> You can find all the available options and the documentation and how to configure them on [the documentation page for the dbt-sqlserver adapter](mssql-profile#connecting-to-azure-sql-server-with-dbt-sqlserver).

Reviewer: ```suggestion
You can find all available options, the documentation, and how to configure them on [the documentation page for the dbt-sqlserver adapter](mssql-profile#connecting-to-azure-sql-server-with-dbt-sqlserver).
```
